motionplan/constraint.go

```diff
@@ -455,3 +456,24 @@ func NewLineConstraint(pt1, pt2 r3.Vector, tolerance float64) (StateConstraint,
 
 	return validFunc, gradFunc
 }
+
+// NewOctreeCollisionConstraint takes an octree and will return a constraint that checks whether any of the geometries in the solver frame
+// intersect with points in the octree. Threshold sets the confidence level required for a point to be considered, and buffer is the
```

Comment:
```suggestion
// intersect with the points in the octree. The threshold sets the confidence level required for a point to be considered a potential collision, and the buffer is the
```

Reply:
Prefer not to make this change; "threshold" and "buffer" are the names of things, not just concepts, so are referred to as names

Reply:
Valid!